functional/FunctionalTest.java

```diff
@@ -3284,6 +3284,230 @@ public static void composeObject_test6() throws Exception {
     }
   }
 
+  /** Test: enableObjectLegalHold(String bucketName, String objectName, String versionId) */
+  public static void enableObjectLegalHold_test() throws Exception {
+    if (!mintEnv) {
+      System.out.println(
+          "Test: enableObjectLegalHold(String bucketName, String objectName, String versionId)");
+    }
+    long startTime = System.currentTimeMillis();
+    String bucketName = getRandomName();
+    String objectName = getRandomName();
+    try {
+      client.makeBucket(bucketName, null, true);
+      try {
+        try (final InputStream is = new ContentInputStream(1 * KB)) {
+          client.putObject(bucketName, objectName, is, new PutObjectOptions(1 * KB, -1));
+        }
+
+        client.enableObjectLegalHold(bucketName, objectName, null);
+        if (!client.isObjectLegalHoldEnabled(bucketName, objectName, null)) {
+          throw new Exception("[FAILED] isObjectLegalHoldEnabled(): expected: true, got: false");
+        }
+        client.disableObjectLegalHold(bucketName, objectName, null);
```

Comment:
After setting `enableObjectLegalHold` we should attempt deleting the object and confirm that fails.

Reply:
This was earlier implemented, but removed as of now as it works with MinIO, but when ran against S3 with versionId as null it succeeds. (deleted the object).
You need specific object version to fail thus it was removed.

Reply:
IMO we can target to validate MinIO implementation